sample/app-ios/app-ios/DecomposeHelpers/StackView.swift

```diff
@@ -17,7 +17,16 @@ struct StackView<T: AnyObject, Content: View>: View {
     var body: some View {
         // iOS 16.0 has an issue with swipe back see https://stackoverflow.com/questions/73978107/incomplete-swipe-back-gesture-causes-navigationpath-mismanagement
         if #available(iOS 16.1, *) {
-            NavigationStack(path: Binding(get: { stack.dropFirst() }, set: { _ in onBack() })) {
+            NavigationStack(
+                path: Binding(
+                    get: { stack.dropFirst() },
+                    set: { updatedPath in
+                        while stack.count > updatedPath.count + 1 {
```

Comment:
Maybe worth adding a `popTo(index)` method to the component and perform the navigation atomically? Currently every component should be activating and destroying one by one.

Reply:
Never mind, I will fix it later.